### PR TITLE
Syndicate radio jammer can now block suit sensors, beacons, teleporters, and GPS

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
The radio jammer now affects the following objects:
- Suit sensors
- Beacons
- Handheld Teleporter
- GPS items

It is also removed from nuclear operatives syndie loadout due to balance concerns.

## Why It's Good For The Game
I found out on SS14 that the syndicate radio jammer goes the extra mile and blocks out a bunch of other signals. Thought it was cool and wanted to add it to SS13. I expect this to affect balance slightly, but I don't think it's enough to justify making this item cost more.

Gameplay effects will include:
- Easier concealment of dead bodies
- Harder escaping via a handheld teleporter

## Changelog
:cl:
add: The syndicate radio jammer can block suit sensors, beacons, teleporters, and GPS
balance: Due to Space Geneva Conventions regulations, nuclear operative teams are forbidden to have radio jammers
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
I don't think it should be purchasable by nukies with this changes, otherwise you can walk up to the captain and kill them, and they can do literally nothing.

Also FWIW I had a "suit sensor jammer" on my todo list but as an entirely separate item

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -382,13 +386,35 @@ effective or pretty fucking useless.
...
+			gps.cut_overlay("working")
+			gps_component.tracking = FALSE
+
+/obj/item/jammer/proc/disable_beacon_trackers(atom/target)
+	for(var/obj/item/beacon/beacon in target.get_all_contents() + target)
+		beacon.turn_off()
+
+/obj/item/jammer/proc/disable_suit_sensors(atom/target)
+	for(var/obj/item/clothing/under/uniform in target.get_all_contents() + target)
+		if((uniform.has_sensor == NO_SENSORS) || !uniform.sensor_mode)
```
Comment (MrMelbert) [inline code/game/objects/items/devices/traitordevices.dm:412]: Calling `get_all_contents` 4 times seems a little egregious when we could just call it once

Comment (timothymtorres) [reply code/game/objects/items/devices/traitordevices.dm]: Refactored with https://github.com/tgstation/tgstation/pull/87931/commits/b53248df96221261f392664ac105c8706d000d16

**Comment by necromanceranne** [comment]
I'll be honest, this is probably too strong, even for traitors, just with the teleporter blocking alone being an entirely passive effect. I think with all of these combined, it becomes too much of a catch all tool for interference and significantly devalues both mechanics that already exist to disrupt some of these effects but require a more proactive attack (EMPs with regards to suit sensors in particular), but just the presence of these tools whatsoever. 

I do find the handtele to be a bit of an obnoxious tool for escaping, but I would think at least attaching some kind of disruption for it to at least an outward attack is better than passively switching all of these options off in an AOE that also pierces walls.

**Comment by timothymtorres** [comment]
I banned it from nuke ops. It's been a while since I've checked but captain's cannot use the hand teleporter while they have the disk. Considering that there are only two of those objects in existence, it's going to be an extremely rare edge case.

**Comment by necromanceranne** [comment]
image

No Tim, they very much can use the hand teleporter along with the disk.

This is me going from Meta engineering to Meta Teleporter Room. With the disk.

It's also not going to be an edge case. The captain is the most important and most significant target on the station, and also the biggest loot pinata as a result. That's not an edge case. This is a factor every time you engage the captain, because he has a hand teleporter in his office and he probably has it on him.

**Comment by Xackii** [comment]
Why remove jammer from the nukes? Most nuke rounds delayed only because captain abuse teleporter to escape to the other end of the station and so on in a circle without a cooldown.

**Comment by necromanceranne** [comment]
Actually, xacxii does actually make a good point. The prior version of radio jammers were entirely reasonable and functional tools for nukies, so losing those entirely is a terrible loss. While a lot of the time, war nukies will simply sabotage telecomms or just go as fast as possible to outpace any callouts, it can be a major hurdle for the crew to spread the word of their approach during stealth if a few of the nukies have this item. It's effects don't affect the nukies themselves either, their headsets work through the jammer.

**Comment by MrMelbert** [comment]
> Why remove jammer from the nukes? Most nuke rounds delayed only because captain abuse teleporter to escape to the other end of the station and so on in a circle without a cooldown.

Nukie skill issue? Blow up the teleporters?

**Comment by timothymtorres** [comment]
> No Tim, they very much can use the hand teleporter along with the disk.

Good to know but it used to be [restricted](https://github.com/tgstation/tgstation/pull/3275). Dunno when it was changed. Even without my changes, the teleporter is not a guaranteed escape since they can just quickly follow you through the portal.

**Comment by EnterTheJake** [comment]
I like the other buffs, teleporter blocking seems a bit excessive though.

**Comment by necromanceranne** [comment]
> The only thing they don't do is directly disable hand teleporters because that object is so old in the codebase it doesn't have the machinery code to support it.

You're right to point out the EMP, because I was also using that as my example. But the difference between these two effects is stark when you consider that the EMP is both 

A) an attack that is direct and proactive, 
B) leaves evidence of the attack and potentially could alert people around the victim, unless we consider 
C) it's coming from the EMP flashlight, which is still an attack and still requires physical access and still gives the victim an alert that the attack is taking place.

I've a whole lot more of a problem with this being an AOE passive effect than the actual effects themselves, because where it exists in other places, it comes with limitations that require some forethought to their use. Some people do already consider EMPs a problem, I consider them mostly balanced by their potential to backfire on the user, this has none of those potential issues, so this I feel is just unnecessary powercreep on tools already available.

You could instead just make hand teleporters EMP vulnerable. That's as easy as just giving it an ``emp_act()`` proc.

**Comment by Loafin34** [comment]
the EMP_act could be the hand teles slice off a random limb (not head) when you walk through it. seems debilitating enough and also accurate to portals

**Comment by ElGitificador** [comment]
> > Why remove jammer from the nukes? Most nuke rounds delayed only because captain abuse teleporter to escape to the other end of the station and so on in a circle without a cooldown.
> 
> Nukie skill issue? Blow up the teleporters?

i believe hand tele can teleport to any beacon a teleporter has locked into even if the teleporter is gone, or at least if the hub is gone. Also think that captain shouldnt rely so much on hand tele, especially since "fighting" against hand tele is quite frustrating imo. one right click, one left click and now your target is on the other edge of the galaxy. Otherwise cool buffs for an underused item and stealth in general.

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]
I'm okay with this under one condition:
- This is only a thing if you EMAG the jammer

**Comment by MrMelbert** [comment]
> i believe hand tele can teleport to any beacon a teleporter has locked into even if the teleporter is gone, or at least if the hub is gone. 

All you need to do is destroy / disassemble the computer and you can no longer teleport to that destination - even unscrewing the screen counts

> Also think that captain shouldnt rely so much on hand tele

Captain needs something to work with against nuke ops. Otherwise all a captain is, is a loot piniata. Walls and airlocks won't stop them, you can't outrun them, and you can't on your own you certainly can't kill them with shielded hardsuits.

**Comment by MrMelbert** [comment]
> captain is supposed to fight back to back with the crew, but there ve been uncountable rounds where the cap just goes tping from one edge of the station to the other one alone. I m not saying captains wouldnt suffer without it but i find this kind of gameplay quite unhealthy

Unfortunately this is a result of nukies being powercrept over the years, but I digress as we're getting far off topic from the pr

**Comment by ATH1909** [comment]
if someone's watching suit sensors, won't multiple members of X department suddenly dropping off of suit sensors give away that you're using a radio jammer there?

**Comment by xzero314** [comment]
Ah yes just what the game needs. EVEN MORE buffs to NRP wordless kill capability. Captains teleport absolutely should not be jammable or empable.

**Comment by SomeoneYouProbablyKnow** [comment]
In my opinion disabling the hand tele's "teleport to location" function is fine. Beong able to jump halfway across the station is stupid. As long as you can still use the "random location" the hand tele still works as an escape tool.

> the EMP_act could be the hand teles slice off a random limb (not head) when you walk through it. seems debilitating enough and also accurate to portals

This solution runs the risk of turning this into another belt defib situation. Portal that delibs on demant that's quite the weapon.

And I dont like pinning so many "counter" effects on EMP (even though i dont think "counters" work as an idea for ss13).
Kinda emag_act all over again from an even more general effect

> if someone's watching suit sensors, won't multiple members of X department suddenly dropping off of suit sensors give away that you're using a radio jammer there, ironically impeding your stealthy

Good. Fuck suit sensor watchers, just glued to that console all the time for a sniff of a single valid instead of actually playing the funny spess game. Fuck suit sensors as a system really for promoting that behavior. 

But stealth that's pulled off without a hitch is only fun for one person. If you KNOW theres a stealthy enemy on the station now everyone gets to participate even if it's virtually impossible to catch a properly sneaky antag.

**Comment by Livrah** [comment]
> > Why remove jammer from the nukes? Most nuke rounds delayed only because captain abuse teleporter to escape to the other end of the station and so on in a circle without a cooldown.
> 
> Nukie skill issue? Blow up the teleporters?

I built 4 extra behind a wall

**Comment by moocowswag** [comment]
Might aswell also block remote signalers and ntnet for consistency (counter to circuits)

**Comment by moocowswag** [comment]
> I'm okay with this under one condition:
> 
> * This is only a thing if you EMAG the jammer

If this is meant to be a compromise to allow nukies to use radio jammers without the bonus effects, nukies have access to emag via saboteur borg

**Comment by CRITAWAKETS** [comment]
> Captain needs _something_ to work with against nuke ops. Otherwise all a captain is, is a loot piniata. Walls and airlocks won't stop them, you can't outrun them, and on your own you certainly can't kill them with shielded hardsuits.

Honestly that's also all the reason to have this not available to traitors. It turns the captain into a loot pinata for any traitor that just brings a gun and a jammer.

**Comment by ZephyrTFA** [comment]
> > I'm okay with this under one condition:
> > 
> > * This is only a thing if you EMAG the jammer
> 
> If this is meant to be a compromise to allow nukies to use radio jammers without the bonus effects, nukies have access to emag via saboteur borg
> 
> (It just never gets used because sab borgs dont realize they can emag station borgs for the free stun arm and have those borgs get captain)

Then this means they need to spend 20tc for this functionality.
I am okay with that.

**Comment by necromanceranne** [comment]
> > Captain needs _something_ to work with against nuke ops. Otherwise all a captain is, is a loot piniata. Walls and airlocks won't stop them, you can't outrun them, and on your own you certainly can't kill them with shielded hardsuits.
> 
> Honestly that's also all the reason to have this not available to traitors. It turns the captain into a loot pinata for any traitor that just brings a gun and a jammer.

Literally just whip out an ebow and shoot him down and you've got an easy kill. 10 TC for an ebow, 5 for the jammer, 3 for the doormag if you need it. 2 if you need a dagger to finish him off. With the cap's gear, the rest of the round is a cakewalk.

**Comment by Jacquerel** [comment]
Emagging your own gear to make it stronger is a flavour fail and we shouldn't do it, I hate whenever people suggest this.

**Comment by Fikou** [comment]
the radio jammer is already good i dont think it needs to become the ultimate tool in making sure you dont get fucked by anything ever
i like it being on nuke ops too

**Comment by timothymtorres** [comment]
Linking:
- #89829

Since this discussion is relevant.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Linking:
> 
>     * [Make a craftable version of the radio jammers, and make the traitor radio jammer cost 1 TC instead of 5 #89829](https://github.com/tgstation/tgstation/pull/89829)
> 
> 
> Since this discussion is relevant.

slightly ig, this one massively increases the power of an individual jammer, the PR you listed just makes them more available

**Comment by necromanceranne** [comment]
An increase in availability makes this pr significantly less appealing to me

**Pull request closed without merging**